dev/TeachingTip/InteractionTests/TeachingTipTests.cs

```diff
@@ -0,0 +1,820 @@
+            retVal.Y = (int)Math.Floor(double.Parse(elements.targetWidthTextBlock.GetText()));
+            retVal.Z = (int)Math.Floor(double.Parse(elements.targetHeightTextBlock.GetText()));
+            return retVal;
+        }
+
+        private string GetEffectivePlacement()
+        {
+            if (elements.effectivePlacementTextBlock == null)
+            {
+                Log.Comment("Find the EffectivePlacementTextBlock");
+                elements.effectivePlacementTextBlock = new TextBlock(FindElement.ByName("EffectivePlacementTextBlock"));
+                Verify.IsNotNull(elements.effectivePlacementTextBlock);
+            }
+            return elements.effectivePlacementTextBlock.GetText();
+        }
+
+        // The test UI has a list box which the teaching tip populates with messages about which events have fired and other useful
+        // Debugging info. This method returns the message at the provided index, which helps testing that events were received in
+        // the expected order.
+        private ListBoxItem GetTeachingTipDebugMessage(int index)
+        {
+            if (elements.lstTeachingTipEvents == null)
+            {
+                Log.Comment("Find the lstTeachingTipEvents");
+                elements.lstTeachingTipEvents = new ListBox(FindElement.ByName("lstTeachingTipEvents"));
+                Verify.IsNotNull(elements.lstTeachingTipEvents);
+            }
+            return elements.lstTeachingTipEvents.Items[index];
+        }
+
+        private bool WaitForChecked(CheckBox checkBox, double millisecondsTimeout = 2000, bool throwOnError = true)
+        {
+            return WaitForCheckBoxUpdated(checkBox, ToggleState.On, millisecondsTimeout, throwOnError);
+        }
+
+        private bool WaitForUnchecked(CheckBox checkBox, double millisecondsTimeout = 2000, bool throwOnError = true)
+        {
+            return WaitForCheckBoxUpdated(checkBox, ToggleState.Off, millisecondsTimeout, throwOnError);
+        }
+
+        private bool WaitForCheckBoxUpdated(CheckBox checkBox, ToggleState state, double millisecondsTimeout, bool throwOnError)
+        {
+            Log.Comment(checkBox.Name + " Checked: " + checkBox.ToggleState);
+            if (checkBox.ToggleState == state)
+            {
+                return true;
+            }
+            else
+            {
+                Log.Comment("Waiting for toggle state to change");
+                checkBox.GetToggledWaiter().TryWait(TimeSpan.FromMilliseconds(millisecondsTimeout));
+            }
+            if (checkBox.ToggleState != state)
+            {
+                Log.Warning(checkBox.Name + " value never changed");
+                if (throwOnError)
+                {
+                    throw new WaiterException();
+                }
+                else
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        private int WaitForOffsetUpdated(
```

Comment:
The indentation makes this a little confusing. Indent the params if you are going to split across multiples lines.